### ISSUE TITLE
SerialRunner fails to import TestRunner base class

### ISSUE THREAD

**Issue by codejedi365** [gabrieli/jest-serial-runner#16]
I'm not sure what I am doing wrong with the configuration I'm using but I keep getting a runtime TypeError message when running `jest` in a project configuration.  

```log
> eslint-plugin-testcafe-community@1.2.3 test
> jest

TestRunner = [object Object]
TypeError: Class extends value #<Object> is not a constructor or null
    at Object.<anonymous> (/Users/codejedi365/Development/workbench/workspace/eslint-plugin-testcafe-community/node_modules/jest-serial-runner/index.js:4:28)
    at Module._compile (internal/modules/cjs/loader.js:1072:14)
    at Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
    at newLoader (/Users/codejedi365/Development/workbench/workspace/eslint-plugin-testcafe-community/node_modules/pirates/lib/index.js:104:7)
    at Object.newLoader (/Users/codejedi365/Development/workbench/workspace/eslint-plugin-testcafe-community/node_modules/pirates/lib/index.js:104:7)
    at Module.load (internal/modules/cjs/loader.js:937:32)
    at Function.Module._load (internal/modules/cjs/loader.js:778:12)
    at Module.require (internal/modules/cjs/loader.js:961:19)
    at require (internal/modules/cjs/helpers.js:92:18)
    at requireOrImportModule (/Users/codejedi365/Development/workbench/workspace/eslint-plugin-testcafe-community/node_modules/jest-util/build/requireOrImportModule.js:53:28)


Test Suites: 0 of 7 total
Tests:       0 total
Snapshots:   0 total
Time:        0 s, estimated 12 s
```

My project configuration:
```js
// jest.config.js
module.exports = {
    projects: [
        {
            displayName: "unit-tests",
            testMatch: ["<rootDir>/tests/**/*.test.ts"],
            preset: "ts-jest",
            collectCoverage: true,
            collectCoverageFrom: [
                "lib/**/*.ts",
                "index.ts",
                "!**/node_modules/**"
            ]
        },
        {
            // Tests to run in serial
            displayName: "integration-tests",
            collectCoverage: false,
            preset: "ts-jest",
            runner: "jest-serial-runner",
            testMatch: ["<rootDir>/tests/**/*.serial-test.ts"]
        }
    ],
    testPathIgnorePatterns: ["<rootDir>/tests/fixtures/", "<rootDir>/example/"]
};
```
I suspect it might have something to do with the fact I'm using `ts-jest` but I'm unfamiliar of how this changes the runtime execution.

Project URL: https://github.com/testcafe-community/eslint-plugin-testcafe-community/tree/integration-tests

**Comment by elsasslegend** [comment]
I have the same issue, did you find a fix for that ? (other than the debug flag --runInBand)

**Comment by codejedi365** [comment]
@djflex68, no fix as of now and note, I just opened this issue this morning.  The debug flag does not work for my configuration above because it fails on the initial import of `jest-serial-runner`. 

The `--runInBand` option is not viable due to the projects config and using that option defeats the purpose of including this package. I have more tests that are unaffected by being run in parallel which is the desired state but have very few that must to run in series.  I desire to use the projects config to define 2 different test runner configurations for the optimal runtime.

**Comment by codejedi365** [comment]
@djflex68, I spoke too soon.  It's actually a very simple fix. The problem stems from the incompatibility of ES6 modules and the CommonJS standard within Node.  Likely the `jest-runner` peer dependency updated its babel/CommonJS build implementation and it caused a failure of `require()` for the TestRunner class in this module's `index.js`.  References for fix are:
- [StackOverflow](https://stackoverflow.com/a/40295288/3653630)
- [Node.js Docs](https://nodejs.org/api/packages.html)

The fix is to add a `.default` to the require("jest-runner"). Will push a PR shortly.

**Comment by elsasslegend** [comment]
Nice investigation ! I will give a try on monday.

thanks !

**Comment by codejedi365** [comment]
@djflex68, I have a newer update now.  My fix above works for `jest-runner@>=27.x`.  However, their is a peer dependency entry in the this project of `jest-runner@24.9.0`.  My understanding of peer dependency installations is a bit vague but either way I ran into a situation where the dependencies were not resolved to the same and deduped and so the TypeError re-appeared.

Recommending 2 updates, the gist:
```json5
// Current version should be limited to, PR #18
{
  "version": "1.2.1"
  "peerDependencies": {
    "jest-runner": "24.x - 26.x"
  }
}
```
```json5
// New PR #17 for support for 27, breaking change
{
  "version": "2.0.0"
  "peerDependencies": {
    "jest-runner": ">=27.x"
  }
}
```
I also recommend deprecation of `v1.2.0` since it will cause issues now that it no longer supports `>=24.8`.

### Workaround
Published package temporarily which supports `jest@27` out of the box.  
- https://www.npmjs.com/package/@codejedi365/jest-serial-runner
*NOTE: you will need to update your `jest.config.js` where the runner is specified to include the package scope (i.e. full package name).*

**Comment by elsasslegend** [comment]
nice work thanks !

**Comment by dzcpy** [comment]
any updates?

**Comment by yf-hk** [comment]
@gabrieli Could you please approve the PR? Thanks!

**Comment by stubbzilla8** [comment]
I'm running into this issue as well :(

**Comment by stubbzilla8** [comment]
> ### Workaround
> Published package temporarily which supports `jest@27` out of the box.
> 
> * https://www.npmjs.com/package/@codejedi365/jest-serial-runner
>   _NOTE: you will need to update your `jest.config.js` where the runner is specified to include the package scope (i.e. full package name)._

Shout out to codejedi365 though...

npm install @codejedi365/jest-serial-runner --save-dev
and update your jest.config.js runner to have
runner: "@codejedi365/jest-serial-runner" 
in the mean time.

**Comment by AuHau** [comment]
@codejedi365 your fork unfortunatelly does not work for me for the Jest@29.x. Just FYI.

**Comment by AuHau** [comment]
I have forked it and relaxed the peerDependency value 😅
https://www.npmjs.com/package/@ethersphere/jest-serial-runner

**Comment by codejedi365** [comment]
> @codejedi365 your fork unfortunatelly does not work for me for the Jest@29.x. Just FYI.

Thanks for the notification. And the hard work of debugging. I'll take a look and pull in your fork and republish if that resolves everything on the weekend. I'm surprised `>=27.x` doesn't enable `jest@29`.

**Comment by AuHau** [comment]
Well it does enable it, but you have to manually install `jest-runner@29.x` then it will pick it up. For the mode of "no direct dependency, so install it" it uses the 27.x version. That is NPM behaviour...

**Comment by codejedi365** [comment]
> @codejedi365 your fork unfortunatelly does not work for me for the Jest@29.x. Just FYI.

I have published [`@codejedi365/jest-serial-runner@2.0.1`](https://www.npmjs.com/package/@codejedi365/jest-serial-runner) to extend support for `jest@29`.
